src/ocean/net/http/HttpServer.d

```diff
@@ -0,0 +1,232 @@
+
+        this.log = Log.lookup("ocean.net.http.HttpServer.HttpRequestHandler");
+    }
+
+    /**************************************************************************
+
+        Method that handles the HTTP requests, it will call each
+
+        Params:
+            response = the response message that will be sent to the client
+
+        Returns:
+            HttpResponseCode.OK if the request was successfully handled, or
+            HttpResponseCode.BadRequest in case of an error
+
+    **************************************************************************/
+
+    override public HttpResponseCode handleRequest ( out const(char)[] response )
+    {
+        log.trace("Got HTTP request");
+
+        try
+        {
+            foreach ( handler; this.handlers )
+            {
+                auto status = handler(this.request, response);
+
+                if ( status != HttpResponseCode.NotFound )
+                {
+                    return status;
+                }
+            }
+        }
+        catch ( Exception e )
+        {
+            log.error("Can't handle the request: `{}` `{}` `{}`", this.request.uri_string, this.request.msg_body, e.message);
+            response = e.message;
+
+            return HttpResponseCode.BadRequest;
+        }
+
+        return HttpResponseCode.NotFound;
+    }
+
+    /**************************************************************************
+
+        Tells the request message body length.
+        This method should be overridden when a request message body is
+        expected. It is invoked when the message header is completely parsed.
+        The default behavior is expecting no request message body.
+
+        Returns:
+            the request message body length in bytes (0 indicates that no
+            request message body is expected)
+
+     **************************************************************************/
+
+    override protected size_t request_msg_body_length ( )
+    {
+        size_t size;
+
+        try
+        {
+            this.request.getUnsigned("Content-Length", size);
+        }
+        catch(Exception e)
+        {
+            log.error("{}", e.toString);
+        }
+
+        return size;
+    }
+}
+
+/// Usage example for the HttpServer class
+unittest
+{
+    HttpResponseCode iDoNothing( HttpRequest, out const(char)[] )
+    {
+        return HttpResponseCode.NotFound;
+    }
+
+    HttpResponseCode iDoSomething( HttpRequest, out const(char)[] response )
+    {
+        response = "I did something".dup;
+        return HttpResponseCode.OK;
+    }
+
+    auto config = HttpConfig("0.0.0.0", 8080);
+    auto server = new HttpServer([ &iDoNothing, &iDoSomething ]);
+
+    // The following line registers the server with the dispatcher, which in
+    // effect makes it operational, subject to the latter's scheduling.
+    // (Commented out here to not start a server instance in a unit test.)
+    // server.start(config);
```

Comment:
Just use a nested function which is never called:
```
unittest
{
    void main ()
    {
        /// Magic
    }
}
```

Reply:
I tried to do something similar to this:

https://github.com/sociomantic-tsunami/ocean/blob/v5.1.0/src/ocean/util/prometheus/server/PrometheusListener.d#L138

which looked good enough to be merged. Thanks for the tip!